nss-lean-anteaters/nutshell
# Friends Section

## https://github.com/nss-lean-anteaters/nutshell/issues/7

Closes # https://github.com/nss-lean-anteaters/nutshell/issues/7()

## Added a new method to search through users and populate the matches into the DOM when searched.

- 

## Steps to Test

```
git fetch --all
git checkout hl-friends

```

Run the API server from the api file on src level - json-server -p 8088 -w database.json 

Run http-server from dist folder and open in the browser.
Type in "Ron" in the input field and hit search.
The DOM should populate all the users that match the input value.

## Impacted Areas in Application
modified:   api/database.json
modified:   index.html
modified:   scripts/DOM/Utility/header.js
modified:   scripts/DOM/Utility/tagBuilder.js
deleted:    scripts/DOM/friends/addByClick.js
modified:   scripts/DOM/friends/displayFriends.js
modified:   scripts/api/apiManager.js
modified:   scripts/dom/main.js
deleted:    scripts/tagBuilder.js
modified:   styles/styles.css
added:   scripts/DOM/friends/addByClickEvent.js
added:   scripts/DOM/friends/addByClickInput.js

## Definition of Done

1. The project must be fully documented. This includes the following:
    1. Complete README that documents the steps to install the code, how to install any dependencies, or system configuration needed.
    1. Every class must be documented with purpose, author, and methods.
    1. Every method must be documented with purpose and argument list - which itself must contain a short purpose for each argument.
1. The project must be able to run fully, and without errors, on each teammate's system.
1. Fulfills every requirement.
Every line of code has been peer reviewed.
1. For projects that require unit testing, core functionality must be identified and have at least one test for each.
